File: frontend/src/domain/entities/company.ts
export interface Company {
  id: string;
  name: string;
  note: string | null;
  created_at: string;
  updated_at: string;
}

export interface CompanyCreate {
  name: string;
  note?: string;
}

export interface CompanyUpdate {
  name?: string;
  note?: string;
}

File: backend/app/application/dto/company.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class CompanyCreate(BaseModel):
    name: str
    note: str | None = None


class CompanyUpdate(BaseModel):
    name: str | None = None
    note: str | None = None


class CompanyResponse(BaseModel):
    id: UUID
    name: str
    note: str | None
    created_at: datetime
    updated_at: datetime

